feature_engine/datetime/date.py

```diff
@@ -0,0 +1,169 @@
+from feature_engine.datetime.base_transformer import DateTimeBaseTransformer
+from feature_engine.variable_manipulation import _check_input_parameter_variables
+
+
+class ExtractDateFeatures(DateTimeBaseTransformer):
+    """
+    ExtractDateFeatures extract various date-related features from
+    datetime features present in the dataset, adding new columns
+    accordingly.
+
+    **Notes**
+    Day of the month extraction isn't supported by panda modules.
+    This transformer implements a workaround that counts how many blocks
+    of 7 days have passed in a given month.
+    A more serious approach based on the calendar should be implemented
+    ***
+
+    Parameters
+    ----------
+    variables: list, default=None
+        The list of variables to impute. If None, the imputer will find and
+        select all datetime variables, including those converted from
+        features of type object/category.
+
+    features_to_extract: list, default = "year"
+        The list of date features to extract. See supported attribute
+
+    drop_datetime: bool, default = "True"
+        Whether to drop datetime variables from the dataframe, including
+        those that have been converted by the transformer.
+        Note: if you pass a subset of features via the variables argument
+        of the transformer and drop_datetime=True, there might be more
+        datetime features in the dataframe that will not be dropped
+        upon calling the transform method
+
+    kwargs:
+        see pd.to_datetime keyword arguments other than "errors"
+
+    Attributes
+    ----------
+    supported:
+        list of supported date features
+
+    Available upon calling the fit method:
+
+    variables_:
+        List of variables from which to extract date features
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        Find the variables from which to extract date features
+    transform:
+        Add the new date features specified in features_to_extract argument
+    fit_transform:
+        Fit to the data, then transform it.
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        features_to_extract: Union[str, List[str]] = "year",
+        drop_datetime: bool = True,
+        **kwargs
+    ) -> None:
+
+        # get the list of supported features from a const variable somewhere?
+        self.supported = [
+            "month",
+            "quarter",
+            "semester",
+            "year",
+            "week_of_the_year",
+            "day_of_the_week",
+            "day_of_the_month",
+            "is_weekend",
+            "week_of_the_month",
+        ]
+
+        if not isinstance(features_to_extract, (str, list)):
+            raise TypeError("features_to_extract must be either a str or a list of str")
+
+        if features_to_extract == "all":
```

Comment:
as per scikit-learn conventions, we should leave the parameters to whatever values are entered by the user (otherwise, the check_estimator tests from sklearn will fail). This bit of code should go in the fit section, and it would look like this:

```
if self.features_to_extract is None:
   self.features_to_extract_ = supported
```

Note the trailing underscore, all parameters or attributes assigned during fit should have a trailing underscore as per sklearn conventions as well. More on this later